api/src/main/java/jakarta/data/metamodel/Pattern.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Operator;
+
+
+/**
+ * Represents a pattern-based restriction for matching operations, encapsulating different
+ * options such as prefix, suffix, and substring matching. This implementation
+ * allows flexibility in creating pattern-based conditions directly as `Restriction` instances.
+ *
+ * <p>Example usage with metadata attributes:</p>
+ * <pre>
+ * // Case-sensitive exact match
+ * Restriction<Book> exactMatch = Pattern.like(_Book.title, "Jakarta Data");
+ *
+ * // Case-insensitive prefix match
+ * Restriction<Book> prefixIgnoreCase = Pattern.prefixedIgnoreCase(_Book.title, "Jak");
+ *
+ * // Case-sensitive suffix match
+ * Restriction<Book> suffixMatch = Pattern.suffixed(_Book.title, "Guide");
+ *
+ * // Case-insensitive substring match
+ * Restriction<Book> substringIgnoreCase = Pattern.substringedIgnoreCase(_Book.title, "Java");
+ * </pre>
+ *
+ * @param <T> the type of the entity on which the restriction is applied.
+ */
+public record Pattern<T>(String field, String value, boolean ignoreCase) implements BasicRestriction<T> {
+
+    /**
+     * The operator for pattern-based restrictions, fixed as `LIKE`.
+     *
+     * @return the `LIKE` operator.
+     */
+    @Override
+    public Operator operator() {
+        return Operator.LIKE;
+    }
+
+    /**
+     * The value of the pattern used for matching.
+     *
+     * @return the pattern value.
+     */
+    @Override
+    public String value() {
+        return value;
+    }
+
+    /**
+     * Creates a pattern for an exact `LIKE` match on the specified value.
+     * This method sets the field to `null`, allowing it to be applied
+     * later to a specific attribute.
+     *
+     * <p>Example usage:</p>
+     * <pre>
+     * Restriction<Book> titlePattern = _Book.title.like(Pattern.like("Jakarta"));
```

Comment:
Bad practice to use like for equals. Adding a wildcard will make it a better example,
```suggestion
     * Restriction<Book> titlePattern = _Book.title.like(Pattern.like("Jakarta %"));
```